File: sabq app ios/sabq/Screens/WorldCupHub.swift
import SwiftUI

// MARK: - أقسام مكمّلة لمونديال 2026 (تكافؤ مع صفحة الويب /world-cup)
//
// ثلاثة أقسام عامة تُستهلك من /api/world-cup/*: حقائق البطولة، شجرة الأدوار
// الإقصائية، وآخر أخبار المونديال. كلها تتبع نمط ZStack+Color.clear حتى تبقى
// الحاوية حيّة فلا يسقط معدّل .task قبل وصول البيانات (فخ Group/EmptyView).

// MARK: - حقائق البطولة (/world-cup/facts)

struct WCFactsSection: View {
    @State private var facts: WCCompetitionFacts?

    var body: some View {
        ZStack(alignment: .topLeading) {
            Color.clear.frame(width: 0, height: 0)
            if let f = facts, f.hasContent {
                content(f)
            }
        }
        .task {
            if let r = try? await APIClient.shared.fetchWorldCupFacts() {
                await MainActor.run { facts = r }
            }
        }
    }

    private func content(_ f: WCCompetitionFacts) -> some View {
        VStack(alignment: .leading, spacing: 14) {
            WCSectionHeader(icon: "rosette", title: "حقائق البطولة",
                            subtitle: "أرقام وذاكرة كأس العالم", tint: WCTheme.gold)
                .padding(.horizontal, 16)

            ScrollView(.horizontal, showsIndicators: false) {
                HStack(spacing: 12) {
                    if let champ = f.defendingChampion {
                        factCard(icon: "crown.fill",
                                 caption: "حامل اللقب",
                                 logo: champ.logo,
                                 title: champ.name,
                                 detail: f.defendingChampionTitles.map { "\($0) ألقاب عالمية" })
                    }
                    if let most = f.mostTitles, let top = most.teams.first {
                        factCard(icon: "trophy.fill",
                                 caption: "الأكثر تتويجًا",
                                 logo: top.logo,
                                 title: most.teams.map { $0.name }.joined(separator: " · "),
                                 detail: "\(most.count) ألقاب")
                    }
                    if let host = f.host, !host.isEmpty {
                        factCard(icon: "mappin.and.ellipse",
                                 caption: "الاستضافة",
                                 logo: nil,
                                 title: host,
                                 detail: "مونديال 2026")
                    }
                }
                .padding(.horizontal, 16)
            }
        }
    }

    private func factCard(icon: String, caption: String, logo: String?, title: String, detail: String?) -> some View {
        HStack(spacing: 8) {
            if let logo, !logo.isEmpty {
                WCRemoteImage(url: logo).padding(3).frame(width: 30, height: 30)
                    .background(Circle().fill(.white))
                    .overlay(Circle().stroke(WCTheme.gold.opacity(0.35), lineWidth: 1))
            } else {
                Image(systemName: icon)
                    .font(.system(size: 13, weight: .bold)).foregroundStyle(WCTheme.gold)
                    .frame(width: 30, height: 30)
                    .background(Circle().fill(WCTheme.gold.opacity(0.14)))
            }
            VStack(alignment: .leading, spacing: 1) {
                Text(caption).font(SabqFonts.app(size: 9, weight: .bold)).foregroundStyle(WCTheme.gold)
                Text(title)
                    .font(SabqFonts.app(size: 12, weight: .heavy)).foregroundStyle(WCTheme.onDark)
                    .lineLimit(1).minimumScaleFactor(0.7)
                if let detail {
                    Text(detail).font(SabqFonts.app(size: 9)).foregroundStyle(WCTheme.onDarkDim).lineLimit(1)
                }
            }
        }
        .padding(.horizontal, 10).padding(.vertical, 7)
        .frame(width: 168, alignment: .leading)
        .background(RoundedRectangle(cornerRadius: 13, style: .continuous).fill(WCTheme.card))
        .overlay(RoundedRectangle(cornerRadius: 13, style: .continuous).stroke(WCTheme.gold.opacity(0.20), lineWidth: 0.5))
    }
}

// MARK: - الأدوار الإقصائية (/world-cup/bracket)
//
// شجرة الويب الأفقية لا تناسب شاشة الهاتف، فنعرض اختيار دور أفقيًا ثم
// مباريات الدور المختار كبطاقات (مطابق سلوك الويب على الموبايل). يُختار
// افتراضيًا أول دور فيه مباراة حيّة، ثم قادمة، ثم أول دور غير فارغ.

struct WCKnockoutSection: View {
    let onOpenMatch: (Int) -> Void

    @State private var bracket: WCBracket?
    @State private var userRound: String?   // roundEn المختار يدويًا

    private var rounds: [WCBracketRound] { bracket?.rounds ?? [] }
    private var playableRounds: [WCBracketRound] { rounds.filter { !$0.matches.isEmpty } }

    private var defaultRoundEn: String? {
        if let live = playableRounds.first(where: { $0.matches.contains { $0.status.live } }) { return live.roundEn }
        if let upcoming = playableRounds.first(where: { $0.matches.contains { !$0.status.finished } }) { return upcoming.roundEn }
        return playableRounds.first?.roundEn
    }
    private var selectedEn: String? { userRound ?? defaultRoundEn }
    private var selectedRound: WCBracketRound? {
        playableRounds.first { $0.roundEn == selectedEn }
    }

    var body: some View {
        ZStack(alignment: .topLeading) {
            Color.clear.frame(width: 0, height: 0)
            if bracket != nil {
                if playableRounds.isEmpty {
                    emptyCard
                } else {
                    content
                }
            }
        }
        .task {
            if let r = try? await APIClient.shared.fetchWorldCupBracket() {
                await MainActor.run { bracket = r }
            }
        }
    }

    private var content: some View {
        VStack(alignment: .leading, spacing: 14) {
            WCSectionHeader(icon: "trophy", title: "الأدوار الإقصائية",
                            subtitle: "طريق اللقب من دور الـ32 حتى النهائي")
                .padding(.horizontal, 16)

            ScrollView(.horizontal, showsIndicators: false) {
                HStack(spacing: 8) {
                    ForEach(playableRounds) { round in
                        let active = round.roundEn == selectedEn
                        Button { withAnimation(.easeOut(duration: 0.2)) { userRound = round.roundEn } } label: {
                            Text(round.round)
                                .font(SabqFonts.app(size: 13, weight: .semibold))
                                .foregroundStyle(active ? .white : WCTheme.onDarkDim)
                                .padding(.horizontal, 13).padding(.vertical, 7)
                                .background(Capsule().fill(active ? WCTheme.emeraldDeep : WCTheme.chipFill))
                        }
                        .buttonStyle(.plain)
                    }
                }
                .padding(.horizontal, 16)
            }

            if let round = selectedRound {
                LazyVStack(spacing: 12) {
                    ForEach(round.matches) { f in
                        WCMatchCard(fixture: f) { onOpenMatch(f.id) }
                    }
                }
                .padding(.horizontal, 16)
            }
        }
    }

    // قبل اعتماد القرعة/انطلاق الأدوار — بطاقة معرّفة بالمراحل بدل إخفاء كامل
    private var emptyCard: some View {
        VStack(alignment: .leading, spacing: 14) {
            WCSectionHeader(icon: "trophy", title: "الأدوار الإقصائية",
                            subtitle: "تبدأ بعد اكتمال دور المجموعات")
                .padding(.horizontal, 16)

            VStack(spacing: 10) {
                ForEach(["دور الـ32", "دور الـ16", "دور الـ8", "دور الـ4", "النهائي"], id: \.self) { label in
                    HStack {
                        Image(systemName: "flag.checkered").font(.system(size: 12)).foregroundStyle(WCTheme.gold)
                        Text(label).font(SabqFonts.app(size: 14, weight: .bold)).foregroundStyle(WCTheme.onDark)
                        Spacer()
                        Text("يُحدَّد لاحقًا").font(SabqFonts.app(size: 11)).foregroundStyle(WCTheme.onDarkDim)
                    }
                    .padding(.horizontal, 14).padding(.vertical, 11)
                    .background(RoundedRectangle(cornerRadius: 14, style: .continuous).fill(WCTheme.card))
                }
            }
            .padding(.horizontal, 16)
        }
    }
}

// MARK: - أخبار المونديال (/world-cup/news)
//
// بطاقات تفتح المقال داخل التطبيق عبر ArticleSlugRoute. تُخفى كليًا عند
// غياب الأخبار. الصورة بنقطة تركيز المحرّر (FocalCachedAsyncImage)، وعند
// غيابها تظهر بطاقة بصرية بشعارَي المنتخبين.

struct WCNewsSection: View {
    @State private var news: [WCNewsItem] = []

    var body: some View {
        ZStack(alignment: .topLeading) {
            Color.clear.frame(width: 0, height: 0)
            if !news.isEmpty {
                content
            }
        }
        .task {
            if let r = try? await APIClient.shared.fetchWorldCupNews(limit: 8) {
                await MainActor.run { news = r }
            }
        }
    }

    private var content: some View {
        VStack(alignment: .leading, spacing: 14) {
            WCSectionHeader(icon: "newspaper.fill", title: "أخبار المونديال",
                            subtitle: "آخر مستجدّات كأس العالم 2026")
                .padding(.horizontal, 16)

            LazyVStack(spacing: 12) {
                ForEach(news) { item in
                    NavigationLink(value: ArticleSlugRoute(slug: item.slug)) {
                        newsCard(item)
                    }
                    .buttonStyle(.plain)
                }
            }
            .padding(.horizontal, 16)
        }
    }

    private func newsCard(_ item: WCNewsItem) -> some View {
        VStack(alignment: .leading, spacing: 0) {
            visual(item)
                .frame(height: 168)
                .frame(maxWidth: .infinity)
                .clipped()
                .clipShape(UnevenRoundedRectangle(topLeadingRadius: 18, bottomLeadingRadius: 0,
                                                  bottomTrailingRadius: 0, topTrailingRadius: 18,
                                                  style: .continuous))

            VStack(alignment: .leading, spacing: 8) {
                HStack(spacing: 6) {
                    Text(kindLabel(item.kind))
                        .font(SabqFonts.app(size: 10, weight: .bold)).foregroundStyle(WCTheme.emeraldDeep)
                        .padding(.horizontal, 8).padding(.vertical, 3)
                        .background(Capsule().fill(WCTheme.emerald.opacity(0.14)))
                    Spacer()
                    if let d = item.publishedDate {
                        Text(Self.relativeTime(d))
                            .font(SabqFonts.app(size: 11)).foregroundStyle(WCTheme.onDarkDim)
                    }
                }
                Text(item.title)
                    .font(SabqFonts.app(size: 16, weight: .bold)).foregroundStyle(WCTheme.onDark)
                    .lineLimit(2).multilineTextAlignment(.leading)
                    .frame(maxWidth: .infinity, alignment: .leading)
                if let excerpt = item.excerpt, !excerpt.isEmpty {
                    Text(excerpt)
                        .font(SabqFonts.app(size: 13)).foregroundStyle(WCTheme.onDarkDim)
                        .lineLimit(2).multilineTextAlignment(.leading)
                        .frame(maxWidth: .infinity, alignment: .leading)
                }
            }
            .padding(14)
        }
        .background(RoundedRectangle(cornerRadius: 18, style: .continuous).fill(WCTheme.card))
        .overlay(RoundedRectangle(cornerRadius: 18, style: .continuous).stroke(WCTheme.cardStroke.opacity(0.5), lineWidth: 0.5))
    }

    @ViewBuilder private func visual(_ item: WCNewsItem) -> some View {
        if let urlString = item.imageUrl, !urlString.isEmpty, let url = URL(string: urlString) {
            FocalCachedAsyncImage(
                url: url,
                focalPoint: ImageFocalPoint(rawX: item.imageFocalPoint?.x, rawY: item.imageFocalPoint?.y)
            ) {
                matchupVisual(item)
            }
        } else {
            matchupVisual(item)
        }
    }

    // بديل بصري بشعارَي المنتخبين على تدرّج الملعب عند غياب الصورة.
    private func matchupVisual(_ item: WCNewsItem) -> some View {
        ZStack {
            LinearGradient(colors: [WCTheme.heroTop, WCTheme.heroBottom],
                           startPoint: .topTrailing, endPoint: .bottomLeading)
            if let home = item.home, let away = item.away {
                HStack(spacing: 22) {
                    logoBubble(home.logo)
                    Text("VS").font(SabqFonts.app(size: 16, weight: .black)).foregroundStyle(WCTheme.gold)
                        .environment(\.layoutDirection, .leftToRight)
                    logoBubble(away.logo)
                }
            } else {
                Image(systemName: "trophy.fill").font(.system(size: 40)).foregroundStyle(WCTheme.gold.opacity(0.85))
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
    }

    private func logoBubble(_ url: String) -> some View {
        WCRemoteImage(url: url).padding(8).frame(width: 56, height: 56)
            .background(Circle().fill(.white))
            .overlay(Circle().stroke(.white.opacity(0.4), lineWidth: 1.5))
    }

    private func kindLabel(_ kind: String) -> String {
        switch kind {
        case "preview": return "ما قبل المباراة"
        case "report": return "تقرير المباراة"
        default: return "مونديال 2026"
        }
    }

    /// وقت نسبي عربي بأرقام لاتينية ("منذ 5 دقائق").
    static func relativeTime(_ date: Date) -> String {
        let seconds = max(0, Date().timeIntervalSince(date))
        let minutes = Int(seconds) / 60
        let hours = minutes / 60
        let days = hours / 24
        if minutes < 1 { return "الآن" }
        if minutes < 60 { return "منذ \(minutes) د" }
        if hours < 24 { return "منذ \(hours) س" }
        if days == 1 { return "أمس" }
        if days < 7 { return "منذ \(days) أيام" }
        return WCFormat.dayRiyadh.string(from: date)
    }
}

File: android-native/app/src/main/kotlin/com/sabq/smart/feature/worldcup/WorldCupHub.kt
package com.sabq.smart.feature.worldcup

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material.icons.filled.Flag
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.MilitaryTech
import androidx.compose.material.icons.filled.Newspaper
import androidx.compose.material.icons.filled.WorkspacePremium
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import com.sabq.smart.ui.components.FocalCachedAsyncImage
import com.sabq.smart.ui.components.ImageFocalPoint
import java.time.OffsetDateTime
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.time.format.DecimalStyle
import java.util.Locale

// ---------- حقائق البطولة (/world-cup/facts) ----------

@Composable
fun FactsSection(facts: WcCompetitionFacts?) {
    if (facts == null || !facts.hasContent) return
    Column(verticalArrangement = Arrangement.spacedBy(14.dp), modifier = Modifier.fillMaxWidth()) {
        Box(Modifier.padding(horizontal = 16.dp)) {
            WcSectionHeader(Icons.Filled.MilitaryTech, "حقائق البطولة", "أرقام وذاكرة كأس العالم", tint = WcColors.gold)
        }
        Row(
            horizontalArrangement = Arrangement.spacedBy(12.dp),
            modifier = Modifier.horizontalScroll(rememberScrollState()).padding(horizontal = 16.dp),
        ) {
            facts.defendingChampion?.let { champ ->
                FactCard(
                    icon = Icons.Filled.WorkspacePremium,
                    caption = "حامل اللقب",
                    logo = champ.logo,
                    title = champ.name,
                    detail = facts.defendingChampionTitles?.let { "$it ألقاب عالمية" },
                )
            }
            facts.mostTitles?.takeIf { it.teams.isNotEmpty() }?.let { most ->
                FactCard(
                    icon = Icons.Filled.EmojiEvents,
                    caption = "الأكثر تتويجًا",
                    logo = most.teams.first().logo,
                    title = most.teams.joinToString(" · ") { it.name },
                    detail = "${most.count} ألقاب",
                )
            }
            facts.host?.takeIf { it.isNotBlank() }?.let { host ->
                FactCard(
                    icon = Icons.Filled.LocationOn,
                    caption = "الاستضافة",
                    logo = null,
                    title = host,
                    detail = "مونديال 2026",
                )
            }
        }
    }
}

@Composable
private fun FactCard(
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    caption: String,
    logo: String?,
    title: String,
    detail: String?,
) {
    val shape = RoundedCornerShape(13.dp)
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(8.dp),
        modifier = Modifier.width(168.dp).clip(shape).background(WcColors.card, shape)
            .border(0.5.dp, WcColors.gold.copy(alpha = 0.20f), shape)
            .padding(horizontal = 10.dp, vertical = 7.dp),
    ) {
        if (!logo.isNullOrEmpty()) {
            Box(
                modifier = Modifier.size(30.dp).clip(CircleShape).background(Color.White)
                    .border(1.dp, WcColors.gold.copy(alpha = 0.35f), CircleShape).padding(3.dp),
                contentAlignment = Alignment.Center,
            ) {
                AsyncImage(model = logo, contentDescription = null, contentScale = ContentScale.Fit, modifier = Modifier.fillMaxSize())
            }
        } else {
            Box(
                modifier = Modifier.size(30.dp).clip(CircleShape).background(WcColors.gold.copy(alpha = 0.14f)),
                contentAlignment = Alignment.Center,
            ) {
                Icon(icon, contentDescription = null, tint = WcColors.gold, modifier = Modifier.size(15.dp))
            }
        }
        Column(verticalArrangement = Arrangement.spacedBy(1.dp)) {
            Text(caption, color = WcColors.gold, fontSize = 9.sp, fontWeight = FontWeight.Bold)
            Text(title, color = WcColors.onDark, fontSize = 12.sp, fontWeight = FontWeight.Black, maxLines = 1, overflow = TextOverflow.Ellipsis)
            detail?.let { Text(it, color = WcColors.onDarkDim, fontSize = 9.sp, maxLines = 1) }
        }
    }
}

// ---------- الأدوار الإقصائية (/world-cup/bracket) ----------

@Composable
fun KnockoutSection(bracket: WcBracket?, onOpenMatch: (Int) -> Unit) {
    if (bracket == null) return
    val playable = bracket.rounds.filter { it.matches.isNotEmpty() }

    Column(verticalArrangement = Arrangement.spacedBy(14.dp), modifier = Modifier.fillMaxWidth()) {
        if (playable.isEmpty()) {
            Box(Modifier.padding(horizontal = 16.dp)) {
                WcSectionHeader(Icons.Filled.EmojiEvents, "الأدوار الإقصائية", "تبدأ بعد اكتمال دور المجموعات")
            }
            Column(verticalArrangement = Arrangement.spacedBy(10.dp), modifier = Modifier.padding(horizontal = 16.dp)) {
                listOf("دور الـ32", "دور الـ16", "دور الـ8", "دور الـ4", "النهائي").forEach { label ->
                    Row(
                        verticalAlignment = Alignment.CenterVertically,
                        modifier = Modifier.fillMaxWidth().clip(RoundedCornerShape(14.dp)).background(WcColors.card)
                            .border(0.5.dp, WcColors.cardStroke, RoundedCornerShape(14.dp))
                            .padding(horizontal = 14.dp, vertical = 11.dp),
                    ) {
                        Icon(Icons.Filled.Flag, null, tint = WcColors.gold, modifier = Modifier.size(14.dp))
                        Spacer(Modifier.width(8.dp))
                        Text(label, color = WcColors.onDark, fontSize = 14.sp, fontWeight = FontWeight.Bold)
                        Spacer(Modifier.weight(1f))
                        Text("يُحدَّد لاحقًا", color = WcColors.onDarkDim, fontSize = 11.sp)
                    }
                }
            }
            return@Column
        }

        val defaultRound = playable.firstOrNull { r -> r.matches.any { it.status.live } }?.roundEn
            ?: playable.firstOrNull { r -> r.matches.any { !it.status.finished } }?.roundEn
            ?: playable.first().roundEn
        var userRound by remember { mutableStateOf<String?>(null) }
        val selectedEn = userRound ?: defaultRound
        val selected = playable.firstOrNull { it.roundEn == selectedEn } ?: playable.first()

        Box(Modifier.padding(horizontal = 16.dp)) {
            WcSectionHeader(Icons.Filled.EmojiEvents, "الأدوار الإقصائية", "طريق اللقب من دور الـ32 حتى النهائي")
        }
        Row(
            horizontalArrangement = Arrangement.spacedBy(8.dp),
            modifier = Modifier.horizontalScroll(rememberScrollState()).padding(horizontal = 16.dp),
        ) {
            playable.forEach { round ->
                val active = round.roundEn == selectedEn
                Text(
                    round.round,
                    color = if (active) Color.White else WcColors.onDarkDim,
                    fontSize = 13.sp,
                    fontWeight = FontWeight.SemiBold,
                    modifier = Modifier.clip(RoundedCornerShape(50)).background(if (active) WcColors.emeraldDeep else WcColors.chipFill)
                        .clickable { userRound = round.roundEn }.padding(horizontal = 13.dp, vertical = 7.dp),
                )
            }
        }
        Column(verticalArrangement = Arrangement.spacedBy(12.dp), modifier = Modifier.padding(horizontal = 16.dp)) {
            selected.matches.forEach { f -> WcKnockoutMatchCard(f, onOpenMatch) }
        }
    }
}

@Composable
private fun WcKnockoutMatchCard(f: WcFixture, onOpenMatch: (Int) -> Unit) {
    Column(
        verticalArrangement = Arrangement.spacedBy(10.dp),
        modifier = Modifier.fillMaxWidth().clip(RoundedCornerShape(18.dp)).background(WcColors.card)
            .border(0.5.dp, WcColors.cardStroke, RoundedCornerShape(18.dp))
            .clickable { onOpenMatch(f.id) }.padding(14.dp),
    ) {
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
            Text(f.round, color = WcColors.onDarkDim, fontSize = 11.sp, fontWeight = FontWeight.SemiBold)
            WcStatusPill(f, onDark = false)
        }
        KnockoutTeamRow(f.home, if (f.started) f.goals.home ?: 0 else null, f.home.winner == true)
        KnockoutTeamRow(f.away, if (f.started) f.goals.away ?: 0 else null, f.away.winner == true)
        f.penalties?.let {
            Text("ركلات الترجيح: ${it.home ?: 0} - ${it.away ?: 0}", color = WcColors.onDarkDim, fontSize = 11.sp)
        }
    }
}

@Composable
private fun KnockoutTeamRow(team: WcTeam, goals: Int?, win: Boolean) {
    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp), modifier = Modifier.fillMaxWidth()) {
        WcTeamLogo(team, size = 26)
        Text(team.name, color = WcColors.onDark, fontSize = 14.sp, fontWeight = if (win) FontWeight.Black else FontWeight.SemiBold, modifier = Modifier.weight(1f))
        if (goals != null) {
            Text("$goals", color = if (win) WcColors.emerald else WcColors.onDark, fontSize = 16.sp, fontWeight = FontWeight.Black)
        }
    }
}

// ---------- أخبار المونديال (/world-cup/news) ----------

@Composable
fun NewsSection(news: List<WcNewsItem>, onOpenArticle: (slug: String) -> Unit) {
    if (news.isEmpty()) return
    Column(verticalArrangement = Arrangement.spacedBy(14.dp), modifier = Modifier.fillMaxWidth()) {
        Box(Modifier.padding(horizontal = 16.dp)) {
            WcSectionHeader(Icons.Filled.Newspaper, "أخبار المونديال", "آخر مستجدّات كأس العالم 2026")
        }
        Column(verticalArrangement = Arrangement.spacedBy(12.dp), modifier = Modifier.padding(horizontal = 16.dp)) {
            news.forEach { item -> NewsCard(item) { onOpenArticle(item.slug) } }
        }
    }
}

@Composable
private fun NewsCard(item: WcNewsItem, onClick: () -> Unit) {
    val shape = RoundedCornerShape(18.dp)
    Column(
        modifier = Modifier.fillMaxWidth().clip(shape).background(WcColors.card, shape)
            .border(0.5.dp, WcColors.cardStroke.copy(alpha = 0.5f), shape)
            .clickable(enabled = item.slug.isNotBlank()) { onClick() },
    ) {
        Box(modifier = Modifier.fillMaxWidth().height(168.dp)) {
            if (!item.imageUrl.isNullOrEmpty()) {
                FocalCachedAsyncImage(
                    url = item.imageUrl,
                    focalPoint = ImageFocalPoint.normalised(
                        item.imageFocalPoint?.x?.toFloat(),
                        item.imageFocalPoint?.y?.toFloat(),
                    ),
                    modifier = Modifier.fillMaxSize(),
                    placeholder = { MatchupVisual(item) },
                )
            } else {
                MatchupVisual(item)
            }
        }
        Column(verticalArrangement = Arrangement.spacedBy(8.dp), modifier = Modifier.padding(14.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically, modifier = Modifier.fillMaxWidth()) {
                Text(
                    kindLabel(item.kind),
                    color = WcColors.emeraldDeep, fontSize = 10.sp, fontWeight = FontWeight.Bold,
                    modifier = Modifier.clip(RoundedCornerShape(50)).background(WcColors.emerald.copy(alpha = 0.14f)).padding(horizontal = 8.dp, vertical = 3.dp),
                )
                Spacer(Modifier.weight(1f))
                relativeTime(item.publishedAt)?.let {
                    Text(it, color = WcColors.onDarkDim, fontSize = 11.sp)
                }
            }
            Text(item.title, color = WcColors.onDark, fontSize = 16.sp, fontWeight = FontWeight.Bold, maxLines = 2, overflow = TextOverflow.Ellipsis, lineHeight = 22.sp)
            item.excerpt?.takeIf { it.isNotBlank() }?.let {
                Text(it, color = WcColors.onDarkDim, fontSize = 13.sp, maxLines = 2, overflow = TextOverflow.Ellipsis)
            }
        }
    }
}

@Composable
private fun MatchupVisual(item: WcNewsItem) {
    Box(
        modifier = Modifier.fillMaxSize().background(Brush.linearGradient(listOf(WcColors.heroTop, WcColors.heroBottom))),
        contentAlignment = Alignment.Center,
    ) {
        val home = item.home
        val away = item.away
        if (home != null && away != null) {
            Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(22.dp)) {
                LogoBubble(home.logo)
                Text("VS", color = WcColors.gold, fontSize = 16.sp, fontWeight = FontWeight.Black)
                LogoBubble(away.logo)
            }
        } else {
            Icon(Icons.Filled.EmojiEvents, null, tint = WcColors.gold.copy(alpha = 0.85f), modifier = Modifier.size(40.dp))
        }
    }
}

@Composable
private fun LogoBubble(url: String) {
    Box(
        modifier = Modifier.size(56.dp).clip(CircleShape).background(Color.White)
            .border(1.5.dp, Color.White.copy(alpha = 0.4f), CircleShape).padding(8.dp),
        contentAlignment = Alignment.Center,
    ) {
        AsyncImage(model = url, contentDescription = null, contentScale = ContentScale.Fit, modifier = Modifier.fillMaxSize())
    }
}

private fun kindLabel(kind: String): String = when (kind) {
    "preview" -> "ما قبل المباراة"
    "report" -> "تقرير المباراة"
    else -> "مونديال 2026"
}

private val newsDayFmt: DateTimeFormatter =
    DateTimeFormatter.ofPattern("EEEE، d MMMM", Locale.forLanguageTag("ar"))
        .withDecimalStyle(DecimalStyle.STANDARD).withZone(ZoneId.of("Asia/Riyadh"))

/** وقت نسبي عربي بأرقام لاتينية ("منذ 5 د"). */
private fun relativeTime(iso: String?): String? {
    if (iso.isNullOrBlank()) return null
    val instant = runCatching { OffsetDateTime.parse(iso).toInstant() }.getOrNull() ?: return null
    val seconds = ((System.currentTimeMillis() - instant.toEpochMilli()) / 1000L).coerceAtLeast(0L)
    val minutes = (seconds / 60).toInt()
    val hours = minutes / 60
    val days = hours / 24
    return when {
        minutes < 1 -> "الآن"
        minutes < 60 -> "منذ $minutes د"
        hours < 24 -> "منذ $hours س"
        days == 1 -> "أمس"
        days < 7 -> "منذ $days أيام"
        else -> newsDayFmt.format(instant)
    }
}
